sdk_container/src/third_party/coreos-overlay/sys-apps/ignition/files/0019-detect-multipart-mime-and-ignore-it.patch

```diff
@@ -0,0 +1,42 @@
+From 27a063dcf7dbcbeadd55a35f665fdbb312283d29 Mon Sep 17 00:00:00 2001
+From: Gabriel Adrian Samfira <gsamfira@cloudbasesolutions.com>
+Date: Fri, 19 May 2023 14:00:11 +0300
+Subject: [PATCH 19/19] detect multipart mime and ignore it
+
+Signed-off-by: Gabriel Adrian Samfira <gsamfira@cloudbasesolutions.com>
+---
+ config/util/parsingErrors.go | 11 ++++++++++-
+ 1 file changed, 10 insertions(+), 1 deletion(-)
+
+diff --git a/config/util/parsingErrors.go b/config/util/parsingErrors.go
+index 914bf2bf..ee01481b 100644
+--- a/config/util/parsingErrors.go
++++ b/config/util/parsingErrors.go
+@@ -29,6 +29,15 @@ import (
+ 	"github.com/coreos/vcontext/tree"
+ )
+ 
++func isMultipartMime(userdata []byte) bool {
++	header := strings.SplitN(string(decompressIfGzipped(userdata)), "\n", 2)[0]
++
++	// Trim trailing whitespaces
++	header = strings.TrimRightFunc(header, unicode.IsSpace)
++
++	return strings.Contains(header, "Content-Type: multipart/mixed;")
++}
++
+ func isCloudConfig(userdata []byte) bool {
+ 	header := strings.SplitN(string(decompressIfGzipped(userdata)), "\n", 2)[0]
+ 
+@@ -62,7 +71,7 @@ func decompressIfGzipped(data []byte) []byte {
+ func HandleParseErrors(rawConfig []byte, to interface{}) (report.Report, error) {
+ 	r := report.Report{}
+ 
+-	if isCloudConfig(rawConfig) || isScript(rawConfig) {
++	if isCloudConfig(rawConfig) || isScript(rawConfig) || isMultipartMime(rawConfig) {
```

Comment:
I wonder if it wouldn't be easier to test for a valid ignition config and just ignore foreign formats, instead of failing and dropping to ```emergency.target```. The assumption being that some other tooling might do something meaningful with that data.

Reply:
i don't know how much `isIgnitionConfig` would check, but we wouldn't want ignition to ignore invalid ignition configs and continue booting without error.
generally ignition assumes that a distro that uses ignition relies on it for correctness, and there are only a handful special-cases where it should yield to a different tool (like here). there was also a discussion here: https://github.com/coreos/ignition/issues/849.

Reply:
Apropos, this PR:

  * https://github.com/flatcar/coreos-cloudinit/pull/21

adds multipart mime support to coreos-cloudinit.

Reply:
> I think there is _some_ value in ignoring userdata that is not a valid ignition config.

To clarify, I don't have particularly strong feelings about this. I am okay with just ignoring the extra case proposed in this patch (included in patch 13, as @pothos mentioned).